packages/block-editor/src/components/iframe/index.js

```diff
@@ -293,8 +295,14 @@ function Iframe( {
 		const _src = URL.createObjectURL(
 			new window.Blob( [ html ], { type: 'text/html' } )
 		);
-		return [ _src, () => URL.revokeObjectURL( _src ) ];
-	}, [ html ] );
+		return [
+			_src,
+			() => {
+				URL.revokeObjectURL( _src );
+				resetZoomLevel();
```

Comment:
Should we reset both zoom _and_ the mode? Check out the useZoomOut hook for all the stuff that is related.

Reply:
Yes, thanks for pointing. Included `__unstableSetEditorMode('edit')`.

Reply:
Can someone explain why we're resetting zoom-out in the iframe component? That feels like a random place.

Reply:
I thought `iframe` to be better place because It is handling all zoom-out related calculations when component is mounted + on changes, and cleanup is naturally can go in there when it is unmounted.

Reply:
`useZoomOut` seems to be breaking with this changes (e.g. clicking on browse styles under global styles) triggers `zoom-out` and exits. This is likely because the hook triggered twice in dev mode.

Reply:
I think what's not clear to me is who is responsible for enabling/disabling zoom-out. Is it the "block editor" itself or is it something that the implementation (a button in the UI of the editor ... does). 

Switching to "code editor" is not a "block editor" thing, so sure the iframe is destroyed when you move to block editor but that doesn't mean you moved to "code editor". Maybe it's a "preview iframe". So you just wanted to render something else temporarily...

Reply:
It's both. The design direction indicated a requirement for a manual toggle button. However, there are also scenarios where we want to be "intelligent" and enter/exit room out for certain actions.

I feel Zoom Out is a fairly opinionated state so we shouldn't leave it active if the users moves to do something else.

> Switching to "code editor" is not a "block editor" thing, so sure the iframe is destroyed when you move to block editor but that doesn't mean you moved to "code editor". Maybe it's a "preview iframe". So you just wanted to render something else temporarily...

I'm sorry but I wasn't fully able to follow what you meant here...

Reply:
> I'm sorry but I wasn't fully able to follow what you meant here...

The iFrame is a component that can be used in several context to render "block content". It's shouldn't be related to zoom-out (aside receiving a "scale" prop to adapt). 

Also block-editor has zoom-out state but doesn't have "code editor" state. So if we want to exit zoom-out when we go into the code editor, it should be done at the highest common level (where the code editor state lives)

Reply:
Thanks for elaborating 👍 

My understanding of the code in this PR is that the "zoom out mode" and "zoom level" will be reset when the iframe component is unmounted. This happens as part of a cleanup operation.

Your concern is that this is a blanket solution which will now apply each time the iframe is unmounted.

What you are recommending is that we act specifically to exit zoom out where necessary based on the state that owns the action.

For example:

**Requirement**: exit Zoom Out when switching in/out of "Code Editor" mode.
**Solution**: add necessary code to exit Zoom Out to the `switchEditorMode` action.

**Requirement**: exit Zoom Out when switching between posts / templates in the Editor.
**Solution**: add necessary code to exit Zoom Out to `setEditedEntity` or perhaps `useInitEditedEntityFromURL`?

That seems logical. Rather than implement a blanket solution, allowing the consuming Editor instance to control when to exit Zoom Out rather than forcing on all implementations of block-editor.

Reply:
Yes, thanks for expanding. "unmounting the iframe" seems too broad. As you can unmount the iframe in block previews too when you render and show them... you could argue block previews have their own "block editor state" which is true, but the point here is that the iframe is about rendering the blocks, not controlling which state we're in.